### PR TITLE
Request delete endpoint

### PR DESCRIPTION
**Pull request by simen-rekkedal**
## Related Issue(s)
- #{issue number}

## Verification
- [ ] **Your** code builds clean without any errors or warnings
- [ ] Manual testing done (required)
- [ ] Relevant automated test added (if you find this hard, leave it and we'll help out)
- [ ] All tests run green

## Documentation
- [ ] User documentation is updated with a separate linked PR in [altinn-studio-docs.](https://github.com/Altinn/altinn-studio-docs) (if applicable)

### CODE REVIEW

File: src/Authentication/Controllers/RequestSystemUserController.cs [hunk 1]
```diff
@@ -312,4 +312,21 @@ public async Task<ActionResult<RequestSystemResponse>> RejectSystemUserRequest(i
 
         return NotFound();
     }
+
+    /// <summary>
+    /// Used by the Vendors to delete the chosen Request by guid
+    /// </summary>
+    /// <returns></returns>
+    [Authorize(Policy = AuthzConstants.POLICY_SCOPE_SYSTEMREGISTER_WRITE)]
```
Comment (acn-dgopa) [inline src/Authentication/Controllers/RequestSystemUserController.cs:320]: policy to be updated as my PR is merged

File: src/Authentication/Controllers/RequestSystemUserController.cs [hunk 2]
```diff
@@ -312,4 +312,21 @@ public async Task<ActionResult<RequestSystemResponse>> RejectSystemUserRequest(i
...
+    [HttpDelete("vendor/{requestId}")]
+    public async Task<ActionResult<RequestSystemResponse>> DeleteRequestByRequestId(Guid requestId)
+    {
+        Result<bool> res = await _requestSystemUser.DeleteRequestByRequestId(requestId);
+        if (res.IsProblem)
+        {
```
Comment (acn-dgopa) [inline src/Authentication/Controllers/RequestSystemUserController.cs:326]: I think not a good idea to return forbid when its any type of problem

File: src/Authentication/Controllers/RequestSystemUserController.cs (hunk 1 above)
Comment (simen-rekkedal) [reply src/Authentication/Controllers/RequestSystemUserController.cs]: ok, fixed in new commit

File: src/Authentication/Controllers/RequestSystemUserController.cs (hunk 2 above)
Comment (simen-rekkedal) [reply src/Authentication/Controllers/RequestSystemUserController.cs:326]: ok

File: test/Altinn.Platform.Authentication.Tests/Controllers/RequestControllerTests.cs [hunk 3]
```diff
@@ -695,12 +698,88 @@ public async Task Get_All_Requests_By_System_Paginated_Several()
...
 
-        HttpResponseMessage message3 = await client.GetAsync(res2.Links.Next);
+        HttpResponseMessage message3 = await client2.GetAsync(res2.Links.Next);
         Assert.Equal(HttpStatusCode.OK, message3.StatusCode);
         Paginated<RequestSystemResponse>? res3 = await message3.Content.ReadFromJsonAsync<Paginated<RequestSystemResponse>>();
         Assert.True(res3 is not null);
     }
 
+    [Fact]
+    public async Task Delete_Request_ByGuid_Accepted()
```
Comment (acn-dgopa) [inline test/Altinn.Platform.Authentication.Tests/Controllers/RequestControllerTests.cs:708]: may be can add a test for requestid not found

**Review by acn-dgopa — APPROVED** [review]
if possible, increase the code coverage

File: test/Altinn.Platform.Authentication.Tests/Controllers/RequestControllerTests.cs (hunk 3 above)
Comment (simen-rekkedal) [reply test/Altinn.Platform.Authentication.Tests/Controllers/RequestControllerTests.cs:708]: will do in future housekeeping PR

**Pull request merged**